src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -164,12 +164,25 @@ public static unsafe Task GetAddrInfoAsync(string hostName, bool justAddresses,
             {
                 state.RegisterForCancellation(cancellationToken);
             }
+            else if (errorCode == SocketError.TryAgain)
+            {
+                // WSATRY_AGAIN indicates possible problem with reachability according to docs.
+                // However, if servers are really unreachable, we would still get IOPending here
+                // and final result would be posted via overlapped IO.
+                // synchronous failure here may signal issue when GetAddrInfoExW does not work from
+                // impersonated context.
+                GetAddrInfoExContext.FreeContext(context);
+                task = Task.CompletedTask;
+                return false;
```

Comment:
Did you consider just returning null vs state.Task, rather than also having a bool?

Reply:
I originally did for the TryGetAddrInfoWithTelemetryAsync but not here. 
Since this is not Async I can change return to `Task?`
If I do, should I change the name as the TryXXX seems to consistently return bool.